File: CS/SchedulerMVVMScenarioWpf/SchedulerViewModel.cs
using System;
using System.Linq;
using System.Data.Objects;
using System.Windows.Forms;
using System.Windows.Input;

using DevExpress.XtraScheduler;
using System.Data.Entity;
using System.ComponentModel;
using System.Collections.Generic;
using DevExpress.Mvvm;
using DevExpress.Xpf.Scheduler;

namespace SchedulerMVVMScenarioWpf {
    public class SchedulerViewModel : INotifyPropertyChanged {
        private TimeInterval lastFetchedInterval;
        private CarsXtraSchedulingEntities carsXtraSchedulingEntities;

        private BindingList<CarScheduling> appointments;
        private BindingList<Car> resources;

        public SchedulerViewModel() {
            lastFetchedInterval = new TimeInterval();
            carsXtraSchedulingEntities = new CarsXtraSchedulingEntities();
            carsXtraSchedulingEntities.Cars.Load();
            Resources = carsXtraSchedulingEntities.Cars.Local.ToBindingList();
           
            AppointmentsModifiedCommand = new DelegateCommand< DevExpress.XtraScheduler.PersistentObjectsEventArgs>(AppointmentsModifiedExecute);
            FetchAppointmentsCommand = new DelegateCommand<DevExpress.XtraScheduler.FetchAppointmentsEventArgs>(FetchAppointmentsExecute);
        }

        public event PropertyChangedEventHandler PropertyChanged;

        protected void OnPropertyChanged(String propertyName) {
            if(this.PropertyChanged != null)
                this.PropertyChanged(this, new PropertyChangedEventArgs(propertyName));
        }

        public ICommand AppointmentsModifiedCommand {
            get;
            private set;
        }
        public ICommand FetchAppointmentsCommand {
            get;
            private set;
        }
        public BindingList<CarScheduling> Appointments {
            get {
                return appointments;
            }
            private set {
                appointments = value;
                OnPropertyChanged("Appointments");
            }
        }
        public BindingList<Car> Resources {
            get {
                return resources;
            }
            private set {
                resources = value;
                OnPropertyChanged("Resources");
            }
        }

        private void AppointmentsModifiedExecute( DevExpress.XtraScheduler.PersistentObjectsEventArgs e) {
            carsXtraSchedulingEntities.SaveChanges();
        }

        private void FetchAppointmentsExecute(DevExpress.XtraScheduler.FetchAppointmentsEventArgs e) {
            TimeInterval currentInterval = e.Interval as TimeInterval;

            if (lastFetchedInterval.Contains(currentInterval))
                return;

            lastFetchedInterval = currentInterval;
            var entities = from entity in carsXtraSchedulingEntities.CarSchedulings
                           where entity.EventType > 0 || (entity.StartTime < lastFetchedInterval.End && entity.EndTime > lastFetchedInterval.Start)
                           select entity;
            entities.Load();
            Appointments = carsXtraSchedulingEntities.CarSchedulings.Local.ToBindingList();
        }

       
    }
}

File: VB/SchedulerMVVMScenarioWpf/SchedulerViewModel.vb
Imports System
Imports System.Linq
Imports System.Data.Objects
Imports System.Windows.Forms
Imports System.Windows.Input

Imports DevExpress.XtraScheduler
Imports System.Data.Entity
Imports System.ComponentModel
Imports System.Collections.Generic
Imports DevExpress.Mvvm
Imports DevExpress.Xpf.Scheduler

Namespace SchedulerMVVMScenarioWpf
    Public Class SchedulerViewModel
        Implements INotifyPropertyChanged

        Private lastFetchedInterval As TimeInterval
        Private carsXtraSchedulingEntities As CarsXtraSchedulingEntities


        Private appointments_Renamed As BindingList(Of CarScheduling)

        Private resources_Renamed As BindingList(Of Car)

        Public Sub New()
            lastFetchedInterval = New TimeInterval()
            carsXtraSchedulingEntities = New CarsXtraSchedulingEntities()
            carsXtraSchedulingEntities.Cars.Load()
            Resources = carsXtraSchedulingEntities.Cars.Local.ToBindingList()

            AppointmentsModifiedCommand = New DelegateCommand(Of DevExpress.XtraScheduler.PersistentObjectsEventArgs)(AddressOf AppointmentsModifiedExecute)
            FetchAppointmentsCommand = New DelegateCommand(Of DevExpress.XtraScheduler.FetchAppointmentsEventArgs)(AddressOf FetchAppointmentsExecute)
        End Sub

        Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

        Protected Sub OnPropertyChanged(ByVal propertyName As String)
            RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
        End Sub

        Private privateAppointmentsModifiedCommand As ICommand
        Public Property AppointmentsModifiedCommand() As ICommand
            Get
                Return privateAppointmentsModifiedCommand
            End Get
            Private Set(ByVal value As ICommand)
                privateAppointmentsModifiedCommand = value
            End Set
        End Property
        Private privateFetchAppointmentsCommand As ICommand
        Public Property FetchAppointmentsCommand() As ICommand
            Get
                Return privateFetchAppointmentsCommand
            End Get
            Private Set(ByVal value As ICommand)
                privateFetchAppointmentsCommand = value
            End Set
        End Property
        Public Property Appointments() As BindingList(Of CarScheduling)
            Get
                Return appointments_Renamed
            End Get
            Private Set(ByVal value As BindingList(Of CarScheduling))
                appointments_Renamed = value
                OnPropertyChanged("Appointments")
            End Set
        End Property
        Public Property Resources() As BindingList(Of Car)
            Get
                Return resources_Renamed
            End Get
            Private Set(ByVal value As BindingList(Of Car))
                resources_Renamed = value
                OnPropertyChanged("Resources")
            End Set
        End Property

        Private Sub AppointmentsModifiedExecute(ByVal e As DevExpress.XtraScheduler.PersistentObjectsEventArgs)
            carsXtraSchedulingEntities.SaveChanges()
        End Sub

        Private Sub FetchAppointmentsExecute(ByVal e As DevExpress.XtraScheduler.FetchAppointmentsEventArgs)
            Dim currentInterval As TimeInterval = TryCast(e.Interval, TimeInterval)

            If lastFetchedInterval.Contains(currentInterval) Then
                Return
            End If

            lastFetchedInterval = currentInterval
            Dim entities = From entity In carsXtraSchedulingEntities.CarSchedulings _
                           Where entity.EventType > 0 OrElse (entity.StartTime < lastFetchedInterval.End AndAlso entity.EndTime > lastFetchedInterval.Start) _
                           Select entity
            entities.Load()
            Appointments = carsXtraSchedulingEntities.CarSchedulings.Local.ToBindingList()
        End Sub


    End Class
End Namespace